diff --git a/test/test_timeout.rb b/test/test_timeout.rb
index 57eca3e..09c650e 100644
--- a/test/test_timeout.rb
+++ b/test/test_timeout.rb
@@ -13,15 +13,15 @@ class TestTimeout < Test::Unit::TestCase
   def test_timeout
     @flag = true
     Thread.start {
-      sleep 0.1
+      sleep 2
       @flag = false
     }
-    assert_nothing_raised("[ruby-dev:38319]") do
-      Timeout.timeout(1) {
+    assert_raise(Timeout::Error, "[ruby-dev:38319]") do
+      Timeout.timeout(0.1) {
         nil while @flag
       }
     end
-    assert !@flag, "[ruby-dev:38319]"
+    assert @flag, "[ruby-dev:38319]"
   end
 
   def test_cannot_convert_into_time_interval
